Lib/test/libregrtest/run_workers.py

```diff
@@ -104,9 +104,9 @@ def __init__(self, worker_id: int, runner: "RunWorkers") -> None:
         self.output = runner.output
         self.timeout = runner.worker_timeout
         self.log = runner.log
-        self.test_name = None
-        self.start_time = None
-        self._popen = None
+        self.test_name: TestName | None = None
+        self.start_time: float | None = None
+        self._popen: subprocess.Popen[str] | None = None
```

Comment:
What is a Popen[str]? Popen doesn't work?

Reply:
`Popen` is [declared as a generic class in typeshed](https://github.com/python/typeshed/blob/bbd9dd1c4f596f564542d48bb05b2cc2e2a7a28d/stdlib/subprocess.pyi#L1843-L1847). If a variable `p` is of type `Popen[str]`, it means that `p.stdin.read()` returns a `str`, and `p.stdin.readlines()` returns a list of `str`s, etc. -- if `p` were of type `Popen[bytes]`, `p.stdin.read()` would return `bytes`, and `p.stdine.readlines()` would return `list[bytes]`.

`Popen` does work, but it's inferred by mypy as being `Popen[Any]`, which is much less precise. It's usually best to give mypy as much information as possible; it enables mypy to make much better inferences.